src/app/api/quality_matrix/replication_source.py

```diff
@@ -101,12 +101,12 @@ def queried_missing_properties(
     ).execute()
 
 
-def join_data(data: dict, key: str) -> dict[str, Union[str, dict]]:
-    return {"metadatum": key, "columns": data}
+def build_quality_output(data: dict, key: str) -> QualityOutput:
```

Comment:
remove function and call constructor directly (i.e. inline it)

Reply:
Inlined